### PR TITLE
[MODORDERS-964] Integrate custom fields for purchase orders and purchase order lines

### PR DESCRIPTION
**Pull request by alb3rtino**
## Purpose
This PR integrates the custom fields functionality for purchase orders and purchase order lines. See [MODORDERS-964](https://issues.folio.org/browse/MODORDERS-964).

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] Were any API paths or methods changed, added or removed?
  - [x] Were there any schema changes?
  - [x] Did any of the interface versions change?
  - [x] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by alb3rtino** [comment]
@SerhiiNosko I'd greatly appreciate it if you or your team could take a moment to review this pull request.

**Comment by SerhiiNosko** [comment]
> @SerhiiNosko I'd greatly appreciate it if you or your team could take a moment to review this pull request.

Hi @alb3rtino sure, looking

File: src/main/java/org/folio/config/ApplicationConfig.java [hunk 1]
```diff
@@ -35,16 +36,23 @@
...
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 @Configuration
 @ComponentScan({ "org.folio.verticles", "org.folio.event.listener" })
-@Import({ KafkaConfiguration.class })
+@Import({ KafkaConfiguration.class, org.folio.spring.ApplicationConfig.class })
 public class ApplicationConfig {
 
+  // Override UserService from folio-custom-fields
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/config/ApplicationConfig.java:50]: I think we need some improvement here, for example if we would like to integrate custom-fields to invoices - we also need to implement a service to interact with users that is not desired.
Did you try to use Spring Conditional annotation to exclude unnecessary beans from spring context?
Or you can use ConditionalOnProperty annotation, so if custom-fields are used only in mod-users - need to add property only to mod-users and by default in all other places this service will not be included into context.

Comment (alb3rtino) [reply src/main/java/org/folio/config/ApplicationConfig.java:50]: Yes. This is a workaround that can/should be removed once the overlaying issue is resolved. I think the work needs to be done in `folio-custom-fields` itself as it currently is depending on a implementation of `UserService`. But it is not on me to make such a decision. I have filed [FCFIELDS-45](https://issues.folio.org/browse/FCFIELDS-45) for this.
Its more or less the same with the `CustomFieldsPostDeployVerticle` and [FCFIELDS-42](https://issues.folio.org/browse/FCFIELDS-42).

For now, these are the only options I found that allow us to move forward with [MODORDERS-964](https://issues.folio.org/browse/MODORDERS-964).

Comment (SerhiiNosko) [reply src/main/java/org/folio/config/ApplicationConfig.java:50]: Okay got it, reaching Volaris team, maybe they already have some plans to prioritize these 2 bugs.

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/config/ApplicationConfig.java (hunk 1 above)
Comment (alb3rtino) [reply src/main/java/org/folio/config/ApplicationConfig.java:50]: Thanks!

**Pull request merged**